src/lib/utils/fmt.h

```diff
@@ -0,0 +1,68 @@
+/*
+* (C) 2023 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_UTIL_FMT_H_
+#define BOTAN_UTIL_FMT_H_
+
+#include <botan/types.h>
+#include <sstream>
+#include <string_view>
+#include <string>
+
+namespace Botan {
+
+namespace fmt_detail {
+
+inline void do_fmt(std::ostringstream& oss, std::string_view format)
+   {
+   oss << format;
+   }
+
+template<typename T, typename... Ts>
+void do_fmt(std::ostringstream& oss, std::string_view format, T val, Ts... rest)
+   {
+   size_t i = 0;
+
+   while(i < format.size())
+      {
+      if(format[i] == '{' && format[i + 1] == '}')
+         {
+         oss << val;
+         return do_fmt(oss, format.substr(i + 2), rest...);
+         }
+      else
+         {
+         oss << format[i];
+         }
+
+      i += 1;
+      }
+   }
```

Comment:
There's an off-by-one here, no?

Consider `format = "I will make the sanitizer unhappy {"`. That will cause an out-of bounds read when looking for the closing brace.

```suggestion
   while(i < format.size() - 1)
      {
      if(format[i] == '{' && format[i + 1] == '}')
         {
         oss << val;
         return do_fmt(oss, format.substr(i + 2), rest...);
         }
      else
         {
         oss << format[i];
         }

      i += 1;
      }
   }
```

Reply:
Fixed now, also added tests